### PR TITLE
fix initialization of var rec_var

### PR DESCRIPTION
**Pull request by jedwards4b**
Fixes issue #1670  logic for initializing rec_var was causing multi-dimensional variable initialization to be overwritten.

### CODE REVIEW

**Comment by jedwards4b** [comment]
@edhartnett are these tests failing because you are actively working on them?  All my tests on cheyenne pass. 
Also I have some improvements to the cmake information in libpio.settings - would you like another PR or can I add them to this one?

**Comment by jedwards4b** [comment]
@edwardhartnett

**Comment by edwardhartnett** [comment]
The travis one is the only one you need to care about. But in it many tests in the cmake build are failing...

**Comment by edwardhartnett** [comment]
@jedwards4b do the improvement to the build summary in a separate PR. Let's try and get this one working...

**Comment by edwardhartnett** [comment]
The failures may be from the address sanitizer. That is, there may be a memory leak with your changes.

**Comment by jedwards4b** [comment]
Looking at the test results it looks like the address sanitizer is also used in the autoconf build where all tests are passing.

**Comment by jedwards4b** [comment]
If I build with gnu and openmpi and use the  -fsanitize=address -fno-omit-frame-pointer flags it also fails on my local system, but it also fails with the current master and nothing in the output indicates a problem with my change
103: Indirect leak of 4 byte(s) in 1 object(s) allocated from:
103:     #0 0x2adf6ae16e58 in __interceptor_malloc ../../../../libsanitizer/asan/asan_malloc_linux.cc:144
103:     #1 0x2adf726e7018  (/glade/u/apps/ch/opt/gnu/9.1.0/lib64/libasan.so.5.0.0+0x248018)
103:     #2 0x616000000000  (<unknown module>)

**Comment by edwardhartnett** [comment]
And yet it worked for me...

Weird. I will take a look at this on Monday morning...

**Comment by edwardhartnett** [comment]
@jedwards4b I cleaned up the CI and merged master. Let's see what happens.

File: src/clib/pioc_support.c [hunk 1]
```diff
@@ -2422,7 +2424,9 @@ inq_file_metadata(file_desc_t *file, int ncid, int iotype, int *nvars,
                     (*rec_var)[v] = 0;
 
             }
+	    PLOG((2,"varid %d var_ndims %d nunlimdims %d rec_var %d unlimdimids[0]=%d var_dimids[0]=%d\n",v,var_ndims, nunlimdims, (*rec_var)[v], unlimdimids[0], var_dimids[0]));
```
Comment (edwardhartnett) [inline src/clib/pioc_support.c:2427]: OK, could this PLOG line of code be causing your problems? In it you reference some [0] array locations, perhaps they are not always there and that is what the memory checker is complaining about?

**Comment by edwardhartnett** [comment]
This needs a test as well.

File: tests/cunit/CMakeLists.txt [hunk 2]
```diff
@@ -236,6 +239,10 @@ else ()
     EXECUTABLE ${CMAKE_CURRENT_BINARY_DIR}/test_darray
     NUMPROCS ${AT_LEAST_FOUR_TASKS}
     TIMEOUT ${DEFAULT_TEST_TIMEOUT})
+  add_mpi_test(test_darray_append
```
Comment (edwardhartnett) [inline tests/cunit/CMakeLists.txt:242]: Can you add this to the Makefile.am as well?

**Comment by edwardhartnett** [comment]
@jedwards4b this looks good to go. Do you want me to merge it?

**Pull request merged**